pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -259,6 +278,66 @@ func (i *OsqueryInstance) Exited() <-chan struct{} {
 	return i.errgroup.Exited()
 }
 
+// ReloadKatcExtension can be called on a running osquery instance to reload its KATC extension
+// manager server, to add new KATC tables or update existing KATC tables' configurations
+// without restarting the entire instance.
+func (i *OsqueryInstance) ReloadKatcExtension(ctx context.Context) error {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	if !i.instanceStarted() {
+		return errors.New("instance not started, cannot reload extension")
+	}
+
+	i.emsLock.Lock()
```

Comment:
should this lock before calling instanceStarted()?

Reply:
Moved lock into `instanceStarted`